### PR TITLE
add object storage functions

### PR DESCRIPTION
**Pull request by mamclaughlin**
### Checklist:

* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [ ] Have you linted your code locally prior to submission?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: cmd/objectStorage.go [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+	// Regenerate
+	objStorageS3KeyRegenerate.Flags().StringP("s3-access-key", "s", "", "access key for a given object storage subscription")
+
+	return objStorageCmd
+}
+
+var objStorageCreate = &cobra.Command{
+	Use:   "create",
+	Short: "",
+	Long:  ``,
```
Comment (geek1011) [inline cmd/objectStorage.go]: The short and long docs should be set, and if not, they are already empty by default.

File: cmd/objectStorage.go [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+	Run: func(cmd *cobra.Command, args []string) {
+		objectStoreClusterID, _ := cmd.Flags().GetInt("obj-store-clusterid")
+		label, _ := cmd.Flags().GetString("label")
+
+		objstorage, err := client.ObjectStorage.Create(context.TODO(), objectStoreClusterID, label)
+
+		if err != nil {
+			fmt.Printf("error creating object storage : %v", err)
```
Comment (geek1011) [inline cmd/objectStorage.go]:
```suggestion
			fmt.Printf("error creating object storage : %v\n", err)
```

File: cmd/objectStorage.go [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+			os.Exit(1)
+		}
+
+		fmt.Printf("created object storage - ID : %v", objstorage.ID)
```
Comment (geek1011) [inline cmd/objectStorage.go]:
```suggestion
		fmt.Printf("created object storage - ID : %v\n", objstorage.ID)
```

File: cmd/objectStorage.go [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+	},
+}
+
+var objStorageLabelSet = &cobra.Command{
+	Use:   "label <objectStorageID>",
+	Short: "change the label for object storage subscription",
+	Long:  ``,
```
Comment (geek1011) [inline cmd/objectStorage.go]: Same here.

File: cmd/objectStorage.go [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+		}
+		return nil
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		id, _ := strconv.Atoi(args[0])
+		label, _ := cmd.Flags().GetString("label")
+
+		err := client.ObjectStorage.SetLabel(context.TODO(), id, label)
+		if err != nil {
+			fmt.Printf("error setting label : %v", err)
```
Comment (geek1011) [inline cmd/objectStorage.go]:
```suggestion
			fmt.Printf("error setting label : %v\n", err)
```

File: cmd/objectStorage.go [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+		}
+
+		fmt.Printf("set label on object storage : %v", id)
+	},
+}
+
+var objStorageList = &cobra.Command{
+	Use:   "list",
+	Short: "retrieves a list of active object storage subscriptions",
+	Long:  ``,
```
Comment (geek1011) [inline cmd/objectStorage.go]: And here.

File: cmd/objectStorage.go [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+
+		if strings.ToLower(includeS3) == "no" {
+			options.IncludeS3 = false
+		}
+
+		if len(args) == 0 {
+			objstorage, err := client.ObjectStorage.List(context.TODO(), options)
+
+			if err != nil {
+				fmt.Printf("error getting object storage : %v", err)
```
Comment (geek1011) [inline cmd/objectStorage.go]:
```suggestion
				fmt.Printf("error getting object storage : %v\n", err)
```

File: cmd/objectStorage.go [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+				os.Exit(1)
+			}
+
+			printer.ObjectStorage(objstorage, options)
+		} else {
+			id, _ := strconv.Atoi(args[0])
+			objstorage, err := client.ObjectStorage.Get(context.TODO(), id)
+
+			if err != nil {
+				fmt.Printf("error getting object storage : %v", err)
```
Comment (geek1011) [inline cmd/objectStorage.go]:
```suggestion
				fmt.Printf("error getting object storage : %v\n", err)
```

File: cmd/objectStorage.go [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+			objstorage, err := client.ObjectStorage.List(context.TODO(), options)
+
+			if err != nil {
+				fmt.Printf("error getting object storage : %v", err)
+				os.Exit(1)
+			}
+
+			printer.ObjectStorage(objstorage, options)
+		} else {
+			id, _ := strconv.Atoi(args[0])
```
Comment (geek1011) [inline cmd/objectStorage.go]: This is never checked. The error should either be handled here or checked in the `Args` func.

File: cmd/objectStorage.go [hunk 10]
```diff
@@ -0,0 +1,188 @@
...
+
+			printer.SingleObjectStorage(objstorage, options)
+		}
+	},
+}
+
+var objStorageClusterList = &cobra.Command{
+	Use:   "list_cluster",
+	Short: "retrieves a list of object storage clusters",
+	Long:  ``,
```
Comment (geek1011) [inline cmd/objectStorage.go]: ditto

File: cmd/objectStorage.go [hunk 11]
```diff
@@ -0,0 +1,188 @@
...
+	Run: func(cmd *cobra.Command, args []string) {
+		cluster, err := client.ObjectStorage.ListCluster(context.TODO())
+
+		if err != nil {
+			fmt.Printf("error getting object storage clusters : %v", err)
```
Comment (geek1011) [inline cmd/objectStorage.go]:
```suggestion
			fmt.Printf("error getting object storage clusters : %v\n", err)
```

File: cmd/objectStorage.go [hunk 12]
```diff
@@ -0,0 +1,188 @@
...
+		printer.ObjectStorageClusterList(cluster)
+	},
+}
+
+var objStorageS3KeyRegenerate = &cobra.Command{
+	Use:   "s3key_regenerate <objectStorageID>",
+	Short: "regenerate the S3 API keys of an object storage subscription",
+	Long:  ``,
+	Run: func(cmd *cobra.Command, args []string) {
+		id, _ := strconv.Atoi(args[0])
```
Comment (geek1011) [inline cmd/objectStorage.go]: This isn't checked either.

File: cmd/objectStorage.go [hunk 13]
```diff
@@ -0,0 +1,188 @@
...
+		s3AccessKey, _ := cmd.Flags().GetString("s3-access-key")
+		s3Keys, err := client.ObjectStorage.RegenerateKeys(context.TODO(), id, s3AccessKey)
+
+		if err != nil {
+			fmt.Printf("error regenerating object storage keys : %v", err)
```
Comment (geek1011) [inline cmd/objectStorage.go]:
```suggestion
			fmt.Printf("error regenerating object storage keys : %v\n", err)
```

File: cmd/objectStorage.go [hunk 14]
```diff
@@ -0,0 +1,188 @@
...
+var objStorageDestroy = &cobra.Command{
+	Use:   "destroy <objectStorageID>",
+	Short: "destroys an object storage subscription",
+	Long:  ``,
+	Run: func(cmd *cobra.Command, args []string) {
+		id, _ := strconv.Atoi(args[0])
+		err := client.ObjectStorage.Delete(context.TODO(), id)
+
+		if err != nil {
+			fmt.Printf("error destroying object storage subscription : %v", err)
```
Comment (geek1011) [inline cmd/objectStorage.go]:
```suggestion
			fmt.Printf("error destroying object storage subscription : %v\n", err)
```

File: cmd/objectStorage.go [hunk 15]
```diff
@@ -0,0 +1,188 @@
...
+		printer.ObjStorageS3KeyRegenerate(s3Keys)
+	},
+}
+
+var objStorageDestroy = &cobra.Command{
+	Use:   "destroy <objectStorageID>",
+	Short: "destroys an object storage subscription",
+	Long:  ``,
+	Run: func(cmd *cobra.Command, args []string) {
+		id, _ := strconv.Atoi(args[0])
```
Comment (geek1011) [inline cmd/objectStorage.go]: Here too.

**Comment by ddymko** [comment]
@mamclaughlin 

- Can we create a get call and only allow the list call to just return a list
- change list_cluster to list-cluster
- change s3key_regenerate... to s3key-regenerate
- camelcase things such as objstorage to objStorage
- objStorageDestroy you should validate that we received the objectStorageID. Similar how its done in other spots such as the objStorageLabelSet call
- ^^ also for objStorageS3KeyRegenerate

**Pull request closed without merging**